INPUTCONFIG.md

```diff
@@ -2,41 +2,37 @@
 
 ## Example
 
-    let inputConfig = {
+    const inputConfig = {
         email: {
             defaultValue: '',
-            validator: (value) => typeof value === 'string' && value.length > 0,
-
-            // Must return a promise
-            // Any errors in the async validation should be handled here
-            asyncValidator: (value) => new Promise(resolve) {
-                ... async stuff ...
-
-                // Resolve if valid, reject if invalid
-                resolve();
-            }
+            validator: (value) => typeof value === 'string' && value.length > 0
         }
     };
 
 This inputConfig defines a single email input with a defaultValue, validator, and asyncValidator.
 
 ## API
 
-### `defaultValue` *(Any)* The value your inputs will start out with.
-
-### `validator(value, inputsState, state)`
-
-Function for performing client side validation
+### `defaultValue` *(Any)* The value your input will start out with. 
 
-#### Arguments
+### `validator(value, inputsState, state, dispatch)`
+ 
+Function for performing validation
+ 
+#### Passed arguments
 1. `value` *(Any)* The value of the input being validated
-2. `inputsState` *(Object)* The current state of your inputs
+2. `inputs` *(Object)* The current state of your inputs
 3. `state` *(Object)* The state of the entire redux store
+4. `dispatch` *(Function)* dispatch function from the redux store, for dispatching side-effect actions
 
-#### Returns
-*(Boolean)*|*(Object)*: If boolean, true for valid, false for invalid. If object, represents the new state of the input.
-This can be used to add error information to the input state, such as setting errorText.
-
+#### Must return one of the following:
```

Comment:
Right now it would be a no-op with an error logged. Maybe it should assume false for nil values.